Increment version to 2.0.1-servicing

wtgodbe opened the pull request:
@weshaggard PTAL - is there a StabilizeVersion equivalent I need to set here?

weshaggard commented:
>PTAL - is there a StabilizeVersion equivalent I need to set here?

No. Instead you have to update the package index. See what I did in master https://github.com/dotnet/corefx/commit/ff19733bfdfb17e5a1e2b79936548bc240ce6cf3.

You also need to update the packages.builds file to only build a subset of packages. See https://github.com/dotnet/corefx/blob/release/1.1.0/src/packages.builds as an example.

wtgodbe commented:
Should the baseline versions there also be updated from 4.3.0 to 4.4.0 (i.e. the same changeset as in the commit you referenced)?

weshaggard commented:
Yes the baselines should also be updated to 4.4.0. I actually meant to point you at commit https://github.com/dotnet/corefx/pull/22704/commits/48ddcc885c027351f5519fcb255f801b13997357 which adds the stable versions.

wtgodbe commented:
@weshaggard can you PTAL at my changes to the packageindex? I updated all Baseline versions of 4.3.0 to 4.4.0, all Baseline versions of 2.0.0 to 2.0.1, and added 2.0.1 as a Stable version for NETCore.Platforms & NETCore.Targets.

weshaggard commented on pkg/Microsoft.Private.PackageBaseline/packageIndex.json:
```diff
@@ -212,7 +214,7 @@
         "4.0.1",
         "4.3.0"
```
You need to also add the stable package versions otherwise the baselines will fail.

weshaggard commented on src/packages.builds:
```diff
@@ -15,6 +16,16 @@
+    <Project Include="$(MSBuildThisFileDirectory)..\pkg\Microsoft.Private.PackageBaseline\Microsoft.Private.PackageBaseline.builds">
+      <AdditionalProperties>$(AdditionalProperties)</AdditionalProperties>
+    </Project>
+    <!-- add specific builds / pkgproj's here to include in servicing builds -->
```
I think the only packages we want to build by default are the ones with Private in their name.  See https://github.com/dotnet/versions/blob/master/build-info/dotnet/corefx/release/2.0.0/Latest_Packages.txt. So you can remove the Microsoft.NETCore.Targets and add the other Private root packages (you don't need to explicitly list out the RID versions of those as they are covered in the builds file).

weshaggard commented on pkg/Microsoft.NETCore.Targets/Microsoft.NETCore.Targets.pkgproj:
```diff
@@ -3,7 +3,7 @@
 
   <PropertyGroup>
-    <PackageVersion>2.0.0</PackageVersion>
+    <PackageVersion>2.0.1</PackageVersion>
```
For the packages we aren't building and shipping we shouldn't update the version number yet. Otherwise some packages might end up with a higher version of this package accidentally.

weshaggard commented on pkg/Microsoft.Private.PackageBaseline/packageIndex.json:
```diff
@@ -1243,18 +1244,18 @@
       "StableVersions": [
         "4.0.0",
         "4.3.0",
-        "4.4.0",
```
This stable version should remain.

weshaggard replied:
I just realized what you have matches master but we should have this version in master as well so please keep this version here.

weshaggard commented on src/packages.builds:
```diff
@@ -15,6 +16,22 @@
+    <Project Include="$(MSBuildThisFileDirectory)..\pkg\Microsoft.Private.CoreFx.NETCoreApp\Microsoft.Private.CoreFx.NETCoreApp.builds">
+      <AdditionalProperties>$(AdditionalProperties)</AdditionalProperties>
+    </Project>
+    <Project Include="$(MSBuildThisFileDirectory)..\pkg\Microsoft.Private.CoreFx.UAP\Microsoft.Private.CoreFx.UAP.builds">
```
For release/2.0.0 we don't actually ship UAP stuff so you should exclude this package. (sorry should have noticed that earlier).

weshaggard commented on src/packages.builds:
```diff
@@ -15,6 +16,22 @@
+    <Project Include="$(MSBuildThisFileDirectory)..\pkg\Microsoft.Private.CoreFx.UAP\Microsoft.Private.CoreFx.UAP.builds">
+      <AdditionalProperties>$(AdditionalProperties)</AdditionalProperties>
+    </Project>
+    <Project Include="$(MSBuildThisFileDirectory)..\pkg\Microsoft.Private.PackageBaseline\Microsoft.Private.PackageBaseline.builds">
```
You have this one twice.

weshaggard commented on pkg/NETStandard.Library.NETFramework/NETStandard.Library.NETFramework.pkgproj:
```diff
@@ -3,7 +3,7 @@
 
   <PropertyGroup>
-    <PackageVersion>2.0.0</PackageVersion>
+    <PackageVersion>2.0.1</PackageVersion>
```
This should remain 2.0.0.

weshaggard commented on src/packages.builds:
```diff
@@ -15,6 +16,16 @@
+      <AdditionalProperties>$(AdditionalProperties)</AdditionalProperties>
+    </Project>
+    <!-- add specific builds / pkgproj's here to include in servicing builds -->
+    <Project Include="$(MSBuildThisFileDirectory)..\pkg\Microsoft.Private.CoreFx.NETCoreApp\Microsoft.Private.CoreFx.NETCoreApp.builds">
```
In the pkgproj for this package you need to explicitly set the PackageVersion to 4.4.1 now.
